include/arch/arm/arch.h

```diff
@@ -98,6 +113,37 @@ extern "C" {
 #define MPU_GUARD_ALIGN_AND_SIZE 0
 #endif
 
+/**
+ * @brief Declare the MPU guard alignment and size for a thread stack
+ *        that is using the Floating Point services.
+ *
+ * For threads that are using the Floating Point services under Shared
+ * Registers (CONFIG_FP_SHARING=y) mode, the exception stack frame may
+ * contain both the basic stack frame and the FP caller-saved context,
+ * upon exception entry. Therefore, a wide guard region is required to
+ * guarantee that stack-overflow detection will always be successful.
+ */
+#if defined(CONFIG_FLOAT) && defined(CONFIG_FP_SHARING) \
+	&& defined(CONFIG_MPU_STACK_GUARD)
+#define MPU_GUARD_ALIGN_AND_SIZE_FLOAT CONFIG_MPU_STACK_GUARD_MIN_SIZE_FLOAT
```

Comment:
do we need add the Kconfig MPU_STACK_GUARD_MIN_SIZE_FLOAT? I think the marco "#define MPU_GUARD_ALIGN_AND_SIZE_FLOAT 128" with some adding explanation like "LOG2CEIL(8 * 4 + 18 * 4)" is ok.